src/lib/pk_pad/emsa_pssr/pssr.cpp

```diff
@@ -88,7 +88,7 @@ bool pss_verify(HashFunction& hash,
       return false;
    }
 
-   std::vector<uint8_t> coded = pss_repr;
+   std::vector<uint8_t> coded(pss_repr.begin(), pss_repr.end());
    if(coded.size() < key_bytes) {
```

Comment:
Perhaps add a TODO to rework that implementation as future work. From a quick skim, I feel like we could avoid this allocation/copy for input that is `>= key_bytes`.

Reply:
We have to have at least one copy since we're applying MGF1 to the data. But in master we have two copies which is certainly not necessary. I fixed this up a bit, likely there is more that can be done.